include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1837,100 +1837,46 @@ struct __parallel_find_or_nd_range_tuner<oneapi::dpl::__internal::__device_backe
-                //    2) early exit from sub-group occurred: in this case the state of __found_local will updated in the next group operation (3)
-                __pred(__item_id, __rng_n, __iters_per_work_item, __wgroup_size, __found_local, __brick_tag, __rngs...);
-
-                // 3. Reduce over group: find __dpl_sycl::__minimum (for the __parallel_find_forward_tag),
-                // find __dpl_sycl::__maximum (for the __parallel_find_backward_tag)
-                // or update state with __dpl_sycl::__any_of_group (for the __parallel_or_tag)
-                // inside all our group items
-                if constexpr (__or_tag_check)
-                    __found_local = __dpl_sycl::__any_of_group(__item_id.get_group(), __found_local);
-                else
-                    __found_local = __dpl_sycl::__reduce_over_group(__item_id.get_group(), __found_local,
-                                                                    typename _BrickTag::_LocalResultsReduceOp{});
-
-                // Set local found state value value to global state to have correct result
-                if (__local_idx == 0)
-                {
-                    __result_and_scratch_storage_t::__get_usm_or_buffer_accessor_ptr(__result_acc)[0] = __found_local;
-                }
-            });
-    });
-
-    // Wait and return result
-    return __result_storage.__wait_and_get_value(__event);
-}
+template <typename KernelName, bool __or_tag_check>
+struct __parallel_find_or_impl_one_wg;
 
 // Base pattern for __parallel_or and __parallel_find. The execution depends on tag type _BrickTag.
-template <typename KernelName, bool __or_tag_check, typename _ExecutionPolicy, typename _BrickTag, typename _AtomicType,
-          typename _Predicate, typename... _Ranges>
-_AtomicType
-__parallel_find_or_impl_multiple_wgs(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec,
-                                     _BrickTag __brick_tag, const std::size_t __rng_n, const std::size_t __n_groups,
-                                     const std::size_t __wgroup_size, const _AtomicType __init_value, _Predicate __pred,
-                                     _Ranges&&... __rngs)
-{
-    auto __result = __init_value;
-
-    // Calculate the number of elements to be processed by each work-item.
-    const auto __iters_per_work_item = oneapi::dpl::__internal::__dpl_ceiling_div(__rng_n, __n_groups * __wgroup_size);
-
-    // scope is to copy data back to __result after destruction of temporary sycl:buffer
+template <typename... KernelName, bool __or_tag_check>
+struct __parallel_find_or_impl_one_wg<__internal::__optional_kernel_name<KernelName...>, __or_tag_check>
+{
+    template <typename _ExecutionPolicy, typename _BrickTag, typename __FoundStateType, typename _Predicate,
+              typename... _Ranges>
+    __FoundStateType
+    operator()(_ExecutionPolicy&& __exec, _BrickTag __brick_tag, const std::size_t __rng_n,
```

Comment:
I see that we are removing the device backend tag here and below as a side effect of this change.  I understand that it is not technically necessary as device backend is the only option at this point.  
However, we decided to include it originally.  Is it useful as a form of documentation still (this is my only guess as to why it was originally included)?

Reply:
Originally, I made it myself during tag dispatch implementation. But at fresh view it simple doesn't make sense at all.